lib/Froxlor/Validate/Validate.php

```diff
@@ -62,32 +62,77 @@ public static function validate($str, $fieldname, $pattern = '', $lng = '', $emp
 		\Froxlor\UI\Response::standard_error($lng, $fieldname, $throw_exception);
 	}
 
-	/**
-	 * Checks whether it is a valid ip
-	 *
-	 * @param string $ip
-	 *        	ip-address to check
-	 * @param bool $return_bool
-	 *        	whether to return bool or call \Froxlor\UI\Response::standard_error()
-	 * @param string $lng
-	 *        	index for error-message (if $return_bool is false)
-	 * @param bool $allow_localhost
-	 *        	whether to allow 127.0.0.1
-	 * @param bool $allow_priv
-	 *        	whether to allow private network addresses
-	 * @param bool $allow_cidr
-	 *        	whether to allow CIDR values e.g. 10.10.10.10/16
-	 *        	
-	 * @return string|bool ip address on success, false on failure
-	 */
-	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false)
+    /**
+     * Converts CIDR to a netmask address
+     *
+     * @thx to https://stackoverflow.com/a/5711080/3020926
+     * @param string $cidr
+     *
+     * @return string
+     */
+    public static function cidr2NetmaskAddr ($cidr) {
+
+        $ta = substr ($cidr, strpos ($cidr, '/') + 1) * 1;
+        $netmask = str_split (str_pad (str_pad ('', $ta, '1'), 32, '0'), 8);
+
+        foreach ($netmask as &$element) {
+            $element = bindec ($element);
+        }
+
+        return implode ('.', $netmask);
+    }
+
+    /**
+     * Checks if an $address (IP) is IPv6
+     *
+     * @param $address
+     *
+     * @return bool
+     * @thx https://stackoverflow.com/a/13677565/3020926
+     */
+    public static function is_ipv6($address) {
+        $ipv4_mapped_ipv6 = strpos($address, "::ffff:");
+        return (strpos($address, ":") !== FALSE) &&
+            ($ipv4_mapped_ipv6 === FALSE || $ipv4_mapped_ipv6 != 0);
+    }
+
+    /**
+     * Checks whether it is a valid ip
+     *
+     * @param string $ip
+     *            ip-address to check
+     * @param bool   $return_bool
+     *            whether to return bool or call \Froxlor\UI\Response::standard_error()
+     * @param string $lng
+     *            index for error-message (if $return_bool is false)
+     * @param bool   $allow_localhost
+     *            whether to allow 127.0.0.1
+     * @param bool   $allow_priv
+     *            whether to allow private network addresses
+     * @param bool   $allow_cidr
+     *            whether to allow CIDR values e.g. 10.10.10.10/16
+     * @param bool   $cidr_as_netmask
+     *            whether to format CIDR nodation to netmask notation
+     *
+     * @return string|bool ip address on success, false on failure
+     */
+	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false, $cidr_as_netmask = false)
```

Comment:
please make `$throw_exception` the last parameter as I've mentioned before (it's always the last parameter also on other functions, it's irritating if this is not the last parameter)

Reply:
That means every call needs to be changed. Why is this the proper solution here?

Reply:
I thought I already explained why: it's always the last parameter also on other functions, it's irritating if this is not the last parameter

The `$throw_exception` was introduced to give a correct API-response when used within the API and remain behaviour if used in Frontend.

Also, it's only 8 occurences...and I bet not even half of them need to be adjusted - I just like to keep things consistent

Reply:
Instead of thinking around it in another topic regarding this (introducing new parameters often comes with such questions and can easily be dealt with somewhere else) you make everyone adapt to your quirks. Thats why I question everything here. But w/e, I'll change it.

Reply:
thank you

Reply:
Changed it https://github.com/pquerner/Froxlor/commit/9689afc759e1e79a1c10f28afb8b0eab0e6c3a49